actors/user/src/main/java/org/sunbird/user/actors/IdentifierFreeUpActor.java

```diff
@@ -0,0 +1,106 @@
+import java.util.Map;
+
+
+@ActorConfig(
+        tasks = {"freeUpUserIdentity"},
+        asyncTasks = {}
+)
+public class IdentifierFreeUpActor extends BaseActor {
+
+    private CassandraOperation cassandraOperation = ServiceFactory.getInstance();
+    private Util.DbInfo usrDbInfo = Util.dbInfoMap.get(JsonKey.USER_DB);
+    private Map<String, Object> userDbMap = new HashMap<>();
+    public static final int FIRST_RECORD = 0;
+
+    @Override
+    public void onReceive(Request request) {
+        String id = (String) request.get(JsonKey.ID);
+        List<String> identifiers = (List) request.get(JsonKey.IDENTIFIER);
+        freeUpUserIdentifier(id, identifiers);
+    }
+
+    private Map<String, Object> getUserById(String id) {
+        Response response = cassandraOperation.getRecordById(usrDbInfo.getKeySpace(), usrDbInfo.getTableName(), id);
+        List<Map<String, Object>> responseList = (List) response.get(JsonKey.RESPONSE);
+        if (CollectionUtils.isEmpty(responseList)) {
+            ProjectCommonException.throwClientErrorException(ResponseCode.invalidUserId);
+        }
+        return responseList.get(FIRST_RECORD);
+    }
+
+    private Response processUserAttribute(List<String> identifiers) {
+
+        identifiers.forEach(val -> {
+            if (val.equalsIgnoreCase(JsonKey.EMAIL)) {
+                nullifyEmail();
+            } else if (val.equalsIgnoreCase(JsonKey.PHONE)) {
+                nullifyPhone();
+            }
+        });
+        return updateUser();
+    }
+
+    private void nullifyEmail() {
+        if (isNullifyOperationValid((String) userDbMap.get(JsonKey.EMAIL))) {
+            userDbMap.replace(JsonKey.PREV_USED_EMAIL, userDbMap.get(JsonKey.EMAIL));
+            userDbMap.replace(JsonKey.EMAIL, null);
+            userDbMap.replace(JsonKey.MASKED_EMAIL, null);
+            userDbMap.replace(JsonKey.EMAIL_VERIFIED, false);
+        }
+    }
+
+    private void nullifyPhone() {
+        if (isNullifyOperationValid((String) userDbMap.get(JsonKey.PHONE))) {
+            userDbMap.replace(JsonKey.PREV_USED_PHONE, userDbMap.get(JsonKey.PHONE));
+            userDbMap.replace(JsonKey.PHONE, null);
+            userDbMap.replace(JsonKey.MASKED_PHONE, null);
+            userDbMap.replace(JsonKey.PHONE_VERIFIED, false);
+            userDbMap.replace(JsonKey.COUNTRY_CODE,null);
+        }
+    }
+
+    private Response updateUser() {
+        return cassandraOperation.updateRecord(usrDbInfo.getKeySpace(), usrDbInfo.getTableName(), userDbMap);
+    }
+
+    private void freeUpUserIdentifier(String id, List<String> identifiers) {
+        userDbMap = getUserById(id);
+        Response response = processUserAttribute(identifiers);
+        sender().tell(response, self());
+        ProjectLogger.log(String.format("%s:%s:USER MAP SUCCESSFULLY UPDATED IN CASSANDRA. WITH VALUES  %s", this.getClass().getSimpleName(), "freeUpUserIdentifier", Collections.singleton(userDbMap.toString())), LoggerEnum.INFO.name());
```

Comment:
should not print complete map , it will have user pii data

Reply:
done